### ISSUE TITLE
[bug] Cannot create a conda environment with conan and python 3.10 (solving fails because of pyyaml)

### ISSUE THREAD

**Issue by nvaytet** [conan-io/conan#10759]
### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 20.04
  * Compiler+version:
  * Conan version: 1.45
  * Python version: 3.10

### Steps to reproduce (Include if Applicable)

I am trying to create a `conda` environment that has `conan` with python 3.10, and also installs `pyyaml`.
The minimal `env.yml` file would look like this:
```
name: testenv
channels:
  - conda-forge
dependencies:
  - conan>=1.45.0
  - python=3.10
  - pyyaml
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

Doing `mamba env create -f env.yml` yields

```
Looking for: ["conan[version='>=1.45.0']", 'python=3.10', 'pyyaml']

Encountered problems while solving:
  - package conan-1.45.0-pyhd8ed1ab_0 requires pyyaml >=3.11,<6.0, but none of the providers can be installed
```

It seems that `conan` requires `pyyaml<6.0`, and when looking at the `pyyaml` packages on [anaconda](https://anaconda.org/conda-forge/pyyaml/files), only version 6 has python 3.10 packages. Versions below 6 do not support python 3.10.

Would it be possible to lift the restriction on `pyyaml<6.0`?
Many thanks.

**Comment by memsharded** [comment]
This is not a bug per se, Conan only works with pyyaml < 6, and it is declared as such.

I see some things in the pyyaml changelog that could be breaking changes. Including dropping Py27 support, which should be taken into account. In any case, it probably requires modifying code, so we might not be able to address this until 2.0, lets see.

**Comment by nvaytet** [comment]
Thanks for the reply.
Actually I don't need to specify `pyyaml` explicitly in the conda env file.
Simply having
```
name: testenv
channels:
  - conda-forge
dependencies:
  - conan>=1.45.0
  - python=3.10
```
will fail with the same error.

So it just means that we cannot create a python 3.10 `conda` environment with `conan`.
I will amend the issue title.

**Comment by memsharded** [comment]
Just in case, to be sure: I have tested installing Conan in Python 3.10, and it installs and works fine. It is installing pyyaml 5.4.1.

So this seems an artificial limitation of Conda. Pyyaml 5.4.1 doesn't seem to have major limitations to not work on Python 3.10, so it is just conda that doesn't package it for some reason. Cannot it be requested somehow in the conda ecosystem? It could be just building the package, while jumping from pyyaml 5 to pyyaml 6 is a major change that would require code changes, and possible having to drop support for pyyaml 5.4, and that could easily be breaking for users in not so edge systems.

**Comment by apteryks** [comment]
That's not just a problem for Conda but for any distribution that globally need to ensure the Python packages can work together at the system level, such as GNU Guix or Debian. For what it's worth, the test suite passes with PyYAML 6.0.

**Comment by SimonHeybrock** [comment]
Any update on this problem?

**Comment by memsharded** [comment]
@Apteryks Did you run the test suite with Python 2.7?

That was probably the biggest blocker. We are finally removing Python 2.7 support in 1.49, because of other dependencies with security vulnerabilities, so maybe this is something that we could try now.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/11363 merged, this will be in 1.49 (very soon)
